gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1403 @@
+    assertTrue(response.body().contains("IncludeThisPageToken"));
+    assertTrue(response.body().contains("contextA"));
+    response = LocalDnsHelper.toListResponse(
+        Lists.newArrayList("some", "multiple", "words"), "contextB", "IncludeThisPageToken", false);
+    assertFalse(response.body().contains("IncludeThisPageToken"));
+    assertTrue(response.body().contains("contextB"));
+    response = LocalDnsHelper.toListResponse(
+        Lists.newArrayList("some", "multiple", "words"), "contextC", null, true);
+    assertFalse(response.body().contains("pageToken"));
+    assertTrue(response.body().contains("contextC"));
+  }
+
+  @Test
+  public void testCreateZoneValidation() {
+    ManagedZone minimalZone = copyZone(ZONE1);
+    // no name
+    ManagedZone copy = copyZone(minimalZone);
+    copy.setName(null);
+    LocalDnsHelper.Response response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("entity.managedZone.name"));
+    // no description
+    copy = copyZone(minimalZone);
+    copy.setDescription(null);
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("entity.managedZone.description"));
+    // no dns name
+    copy = copyZone(minimalZone);
+    copy.setDnsName(null);
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("entity.managedZone.dnsName"));
+    // zone name is a number
+    copy = copyZone(minimalZone);
+    copy.setName("123456");
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("entity.managedZone.name"));
+    assertTrue(response.body().contains("Invalid"));
+    // dns name does not end with period
+    copy = copyZone(minimalZone);
+    copy.setDnsName("aaaaaa.com");
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("entity.managedZone.dnsName"));
+    assertTrue(response.body().contains("Invalid"));
+    // dns name is reserved
+    copy = copyZone(minimalZone);
+    copy.setDnsName("com.");
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(400, response.code());
+    assertTrue(response.body().contains("not available to be created."));
+    // empty description should pass
+    copy = copyZone(minimalZone);
+    copy.setDescription("");
+    response = LOCAL_DNS_HELPER.createZone(PROJECT_ID1, copy);
+    assertEquals(200, response.code());
+  }
+
+  @Test
+  public void testCheckListOptions() {
+    // listing zones
+    optionsMap.put("maxResults", "-1");
+    LocalDnsHelper.Response response = LocalDnsHelper.checkListOptions(optionsMap);
+    assertEquals(400, response.code());
```

Comment:
I mentioned asserting something about the response error messages above, but don't do it in cases like this method where it'll take you a lot of time and not yield all that much benefit.